pandas/tests/sparse/test_frame.py

```diff
@@ -1267,6 +1268,33 @@ def test_from_scipy_correct_ordering(spmatrix):
     tm.assert_frame_equal(sdf.to_dense(), expected.to_dense())
 
 
+def test_from_scipy_object_fillna(spmatrix):
+    # GH 16112
+    tm.skip_if_no_package('scipy', max_version='0.19.0')
+
+    arr = np.eye(3)
+    arr[1:, 0] = np.nan
+
+    try:
+        spm = spmatrix(arr)
+        assert spm.dtype == arr.dtype
+    except (TypeError, AssertionError):
+        # If conversion to sparse fails for this spmatrix type and arr.dtype,
+        # then the combination is not currently supported in NumPy, so we
+        # can just skip testing it thoroughly
+        return
+
+    sdf = pd.SparseDataFrame(spm).fillna(-1.0)
+
+    # Returning frame should fill all nan values with -1.0
+    expected = pd.SparseDataFrame({0: {0: 1.0, 1: np.nan, 2: np.nan},
+                                   1: {0: np.nan, 1: 1.0, 2: np.nan},
+                                   2: {0: np.nan, 1: np.nan, 2: 1.0}}
+                                  ).fillna(-1.0)
+
+    tm.assert_frame_equal(sdf.to_dense(), expected.to_dense())
```

Comment:
Ok, does it work if you instead init the expected frame like so:
```py
expected = pd.SparseDataFrame(
    {0: {0: 1.0, 1: np.nan, 2: np.nan},
     1: {1: 1.0},
     2: {2: 1.0}}).fillna(-1.0)
```
I.e. without explicit NaNs. I guess NaNs as values are distinct from _no values_ in sparse matrices. The alternative which would also work (but is obviously worse) is to init spmatrix like:
```py
arr = np.eye(3)
arr[1:, 0] = np.nan

arr[arr == 0] = np.nan
spm = spmatrix(arr)
```
In either case, the positions of defined values vs. non-defined must match.